File: Javascript/Algorithms/409.js
/**
 * @param {string} s
 * @return {number}
 */
var longestPalindrome = function(s) {
  const floor = Math.floor;
  // Build a count of letters (note, need extra space because some
  // non-letter chars follow uppercase ones.)
  let letters = Array(58).fill(0);
  for (let c of s) letters[c.charCodeAt() - 65]++;

  let length = 0, add = 0;
  letters.forEach(count => {
    if (count & 1) {
      add = 1;
      length--;
    }
    length += count;
  });
  return length + add;
};

File: C/Algorithms/409.c
#define CHARS 58

int longestPalindrome(char *s) {
  // Build a count of letters (note, need extra space because some
  // non-letter chars follow uppercase ones.)
  int letters[CHARS] = {0};
  while (*s) {
    letters[*s++ - 65]++;
  }

  int length = 0, add = 0;
  for (int i = 0; i < CHARS; i++) {
    int count = letters[i];
    if (count & 1) {
      add = 1;
      length--;
    }
    length += count;
  }
  return length + add;
}